src/wp-includes/interactivity-api/class-wp-interactivity-api.php

```diff
@@ -0,0 +1,671 @@
+
+					/*
+					 * Extracts the directive prefix to see if there is a server directive
+					 * processor registered for that directive.
+					 */
+					list( $directive_prefix ) = $this->extract_prefix_and_suffix( $attribute_name );
+					if ( array_key_exists( $directive_prefix, self::$directive_processors ) ) {
+						$directives_prefixes[] = $directive_prefix;
+					}
+				}
+
+				/*
+				 * If this is not a void element, it adds it to the tag stack so it can
+				 * process its closing tag and check for unbalanced tags.
+				 */
+				if ( ! $p->is_void() ) {
+					$tag_stack[] = array( $tag_name, $directives_prefixes );
+				}
+			}
+
+			/*
+			 * Sorts the attributes by the order of the `directives_processor` array
+			 * and checks what directives are present in this element. The processing
+			 * order is reversed for tag closers.
+			 */
+			$directives_prefixes = array_intersect(
+				$p->is_tag_closer()
+					? $directive_processor_prefixes_reversed
+					: $directive_processor_prefixes,
+				$directives_prefixes
+			);
+
+			// Executes the directive processors present in this element.
+			foreach ( $directives_prefixes as $directive_prefix ) {
+				$func = is_array( self::$directive_processors[ $directive_prefix ] )
+					? self::$directive_processors[ $directive_prefix ]
+					: array( $this, self::$directive_processors[ $directive_prefix ] );
+				call_user_func_array(
+					$func,
+					array( $p, &$context_stack, &$namespace_stack )
+				);
+			}
+		}
+
+		/*
+		 * It returns the original content if the HTML is unbalanced because
+		 * unbalanced HTML is not safe to process. In that case, the Interactivity
+		 * API runtime will update the HTML on the client side during the hydration.
+		 */
+		return $unbalanced || 0 < count( $tag_stack ) ? $html : $p->get_updated_html();
+	}
+
+	/**
+	 * Evaluates the reference path passed to a directive based on the current
+	 * store namespace, state and context.
+	 *
+	 * @since 6.5.0
+	 *
+	 * @param string|true $directive_value   The directive attribute value string or `true` when it's a boolean attribute.
+	 * @param string      $default_namespace The default namespace to use if none is explicitly defined in the directive
+	 *                                       value.
+	 * @param array|false $context           The current context for evaluating the directive or false if there is no
+	 *                                       context.
+	 * @return mixed|null The result of the evaluation. Null if the reference path doesn't exist.
+	 */
+	private function evaluate( $directive_value, string $default_namespace, $context = false ) {
+		list( $ns, $path ) = $this->extract_directive_value( $directive_value, $default_namespace );
+		if ( empty( $path ) ) {
+			return null;
+		}
+
+		$store = array(
+			'state'   => isset( $this->state_data[ $ns ] ) ? $this->state_data[ $ns ] : array(),
+			'context' => isset( $context[ $ns ] ) ? $context[ $ns ] : array(),
```

Comment:
```suggestion
			'state'   => $this->state_data[ $ns ] ?? array(),
			'context' => $context[ $ns ] ?? array(),
```

Reply:
Ohhh, true. We can use the null coalescing operator now that WP has dropped support for PHP 5.6.

Done in https://github.com/WordPress/wordpress-develop/pull/5953/commits/eab8f288a5e5861214696ed0fe3ef22ac333bcdf.